pkg/piperutils/slices_test.go

```diff
@@ -51,3 +63,30 @@ func TestTrim(t *testing.T) {
 	assert.Contains(t, s, "apple")
 	assert.Contains(t, s, "mango")
 }
+
+func TestSplitAndTrim(t *testing.T) {
+	t.Run("Separator is not space", func(t *testing.T) {
+		// init
+		s := []string{" a", "", "-a-b --c ", "d-e", "f", " f", ""}
+		// test
+		s = SplitAndTrim(s, "-")
```

Comment:
From the test it looks like duplicates are removed, this is not clear by the function name.